js/controller.js

```diff
@@ -18,14 +18,33 @@ import router from 'core/js/router';
 class TrickleController extends Backbone.Controller {
 
   initialize() {
+    this.listenTo(data, {
+      ready: this.onDataReady
+    });
+  }
+
+   async onDataReady() {
+    const trickleConfig = Adapt.config.get('_trickle');
+    if (trickleConfig?._isEnabled === false) return;
+
+    this.setUpEventListeners();
     this.checkIsFinished = _.debounce(this.checkIsFinished, 1);
```

Comment:
Function debounces should always go in the initialize/constructor to be run only once. There's no send in redebouncing this on a second data ready.